real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/RoboBuggyJFrame.java

```diff
@@ -0,0 +1,83 @@
+package com.roboclub.robobuggy.ui;
+
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.LayoutManager;
+import java.awt.event.ComponentEvent;
+import java.awt.event.ComponentListener;
+import java.io.File;
+import java.util.ArrayList;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+
+/*
+ *  This class is for the robobuggy group to have nice control over how Jframe windows look and resize 
+ */
+public class RoboBuggyJFrame extends JFrame  {
+	
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -3566499518806533434L;
+
+	public RoboBuggyJFrame(String title,double widthPercentage,double heightPercentage) {
+		
+		//sets the title based on this frames name
+		this.setTitle(title);
+
+		//set the window close default
+        this.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+
+		//adds the roboclub icon to the top of the window
+		try {
+			this.setIconImage(ImageIO.read(new File("images/rc_logo.png")));
+		} catch (Exception e) {
+			System.out.println("Unable to read icon image!");
+		}
+		
+		//gets the screen size
+		GraphicsDevice gd = GraphicsEnvironment.getLocalGraphicsEnvironment().getDefaultScreenDevice();
+		int screenWidth = gd.getDisplayMode().getWidth();
+		int screenHeight = gd.getDisplayMode().getHeight();
+		double width = widthPercentage*screenWidth;
+		double height = heightPercentage*screenHeight;
+	
+		//makes the window visible 
+		this.setBounds(0, 0, (int)width, (int)height);
+		this.setVisible(true);
+	}
+	
+	ArrayList<ComponentData> components = new ArrayList<ComponentData>();
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1795142035/issues/source?bid=2938665&fileBranchId=3059924#l55)